packages/advanced-logic/specs/payment-network-erc20-proxy-contract-0.1.0-DRAFT.md

```diff
@@ -0,0 +1,241 @@
+# Payment Network - ERC20 - proxy contract
+
+You may be interested in this document if:
+
+- you want to create your own implementation of the Request protocol
+- you are curious enough to dive and see what is under the hood of the Request protocol
+
+Prerequisite: Having read the advanced logic specification (see [here](./advanced-logic-specs-0.1.0-DRAFT.md)).
+
+## Description
+
+This extension allows the payments and the refunds to be made in ERC20 tokens on the Ethereum blockchain.
+The payment is made through a proxy contract. This proxy contract does the ERC20 token transfer on behalf of
+the user. The contract ensures a link between an ERC20 transfer and a request through a `paymentReference`.
+This `paymentReference` consists of the last 8 bytes of a salted hash of the requestId: `last8Bytes(hash(requestId + salt + address))`:
+
+- `requestId` is the id of the request
+- `salt` is a random number with at least 8 bytes of randomness. It must be unique to each request
+- `address` is the payment address for payments, the refund address for refunds
+- `hash()` is a keccak256 hash function
+- `last8Bytes()` take the last 8 bytes
+
+As a payment network, this extension allows to deduce a payment `balance` for the request. (see
+[Interpretation](#Interpretation))
+
+## Contract
+
+The contract contains one function called `transferWithData` which takes 4 arguments:
```

Comment:
I put `transferFromWithReference` in depend if we consider the proxy smart contract could be used for something else